clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -369,24 +393,40 @@ private void maybeTriggerWakeup() {
     }
 
     public void disableWakeups() {
-        wakeupDisabledCount++;
+        lock.lock();
```

Comment:
I could switch to using the object's monitor instead of an explicit lock (Ewen also suggested this previously). It would be a little less verbose.